### PR TITLE
Ticket #2378: Add readonly portfolio senior official page

### LINKED ISSUE #2378: Portfolio Senior Official page
### Story

As an organization member
I want to see who my senior official is
so that I can request it be change or know who to contact if needed.

### Acceptance Criteria

- [ ] In the organization view the senior official side navigation is a clickable link to the [senior official page](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-Designs-CISA?node-id=4802-15596&t=62Sjs4ioPRj7Ef2G-0)
- [ ] page matches the figma in the previous AC
- [ ] the page pulls from the portfolio Senior official (not the SO on a particular domain)
- [ ] page should be non-editable

### Additional Context

_No response_

### Issue Links

blocked by #2377

### PR DESCRIPTION
**Pull request by zandercymatics**
## Ticket

Resolves #2378

## Changes

- Adds a readonly senior official view for portfolios
- Replaces the existing readonly senior official view under domains with our new readonly format
- Fixes a pre-existing bug with the AdminSortFields that didn't allow an admin user to add new senior officials

## Context for reviewers
The content for this PR [resides on this sandbox](https://getgov-bob.app.cloud.gov/admin/registrar/waffleflag/).

#2378 specifies that we should standardize the senior_official view into one (such that we can reuse this later). As such, this PR combines the new senior_official view on the portfolio page with the one on the manage domain page. This means that the manage domain page now has a proper readonly view.

Another thing to note is that this page behaves like the portfolio organization one - meaning that fields will not populate on the page if they are None. This is a parity move, but we can change this across one or both implementations if we want.

If no senior official is found whatsoever, the page has a display for that. This may also need further design iteration.

## Setup
**For devs/design**
Portfolio:
- Navigate to /admin
- Enable the "organization_feature" flag (in the table Waffle Flags)
- Go to the users table
- Go to your user record
- Add the portfolio [Hotel California](https://getgov-bob.app.cloud.gov/organization/) to your user
- Under portfolio roles, add "admin" to your user
- Navigate to home
- Click "Hotel California"
- Click "Senior official" in the menu

Note: if you want to test the "no senior official" view, just remove the senior official from the **portfolio** object and renavigation to that page. Remember to add one back after you are done testing!

Non-org members:
- Disable the organization_feature waffle flag
- Add a senior official to the domain you want to test
- Click manage, then senior official

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [x] Checked that the design translated visually
- [x] Checked behavior
- [x] Checked different states (empty, one, some, error)
- [x] Checked for landmarks, page heading structure, and links
- [x] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [x] Checked keyboard navigability
- [x] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [x] Chrome
  - [ ] Microsoft Edge
  - [x] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

## Screenshots
image
image
(This edge-case view applies to both)
image

### CODE REVIEW

File: src/registrar/admin.py [hunk 1]
```diff
@@ -493,6 +493,8 @@ def changelist_view(self, request, extra_context=None):
     #     return super().change_view(request, object_id, form_url, extra_context=extra_context)
 
 
+# TODO - this should be refactored. This is shared among every class that inherits this,
+# and it breaks the senior_official field because it exists both as model "Contact" and "SeniorOfficial".
 class AdminSortFields:
```
[the remark this answers is not in the document]
Comment (abroddrick) [reply src/registrar/admin.py:498]: Hey, PLEASE don't leave todo's in the code without making a ticket for removing it. Add all this context to a ticket and make a ticket for it so we can track this. Then put the ticket number in the comment

**Comment by AnnaGingle** [comment]
> @AnnaGingle Hey Anna! I can't replicate that issue. If your using windows, I suspect the culprit is your default app for email. If its not set to the "mail" app (such as a browser), it can cause this issue.
> 
> Are you able to open the email link below? [help@get.gov](mailto:help@get.gov)

Nope. It's def my settings. Thanks for double checking!

**Review by AnnaGingle — APPROVED** [review]
LGTM!

File: src/registrar/tests/test_views_domain.py [hunk 2]
```diff
@@ -1192,64 +1192,28 @@ def assert_all_form_fields_have_expected_values(self, form, test_cases, test_for
...
-        self.assertEqual("CIO", self.domain_information.senior_official.title)
-        self.assertEqual("nobody@igorville.gov", self.domain_information.senior_official.email)
+        self.assertContains(so_page, "Apple Tester")
+        self.assertContains(so_page, "CIO")
+        self.assertContains(so_page, "nobody@igorville.gov")
+        self.assertNotContains(so_page, "Save")
 
     @less_console_noise_decorator
-    def test_domain_edit_senior_official_tribal(self):
+    def test_domain_cannot_edit_senior_official_tribal(self):
```
Comment (Matt-Spence) [inline src/registrar/tests/test_views_domain.py:1216]: Why do we only test these when federal or tribal? Testing it for every single possible generic org seems excessive, but why these two specifically?

**Review by Matt-Spence — COMMENTED** [review]
One small question, other than that looks great!

File: src/registrar/tests/test_views_domain.py (hunk 2 above)
Comment (Matt-Spence) [reply src/registrar/tests/test_views_domain.py:1216]: Ah gotcha.

**Review by Matt-Spence — APPROVED** [review]
Great stuff.

**Pull request merged**